src/Adapter/StockManager.php

```diff
@@ -57,4 +59,119 @@ public function isAsmGloballyActivated()
+     * @param $errorState
+     * @param $cancellationState
+     * @return bool
+     */
+    public function updatePhysicalProductQuantity($shopId, $errorState, $cancellationState)
+    {
+        $this->updateReservedProductQuantity($shopId, $errorState, $cancellationState);
+
+        $updatePhysicalQuantityQuery = '
+            UPDATE {table_prefix}stock_available sa
+            SET sa.physical_quantity = sa.quantity + sa.reserved_quantity
+        ';
+        $updatePhysicalQuantityQuery = str_replace('{table_prefix}', _DB_PREFIX_, $updatePhysicalQuantityQuery);
+
+        return Db::getInstance()->execute($updatePhysicalQuantityQuery);
+    }
+
+    /**
+     * @param $shopId
+     * @param $errorState
+     * @param $cancellationState
+     * @return bool
+     */
+    private function updateReservedProductQuantity($shopId, $errorState, $cancellationState)
+    {
+        $updateReservedQuantityQuery = '
+            UPDATE {table_prefix}stock_available sa
+            SET sa.reserved_quantity = (
+                SELECT SUM(od.product_quantity - od.product_quantity_refunded)
+                FROM {table_prefix}orders o,
+                {table_prefix}order_history oh,
+                {table_prefix}order_state os,
+                {table_prefix}order_detail od
+                WHERE
+                o.id_order = oh.id_order AND
+                o.current_state = oh.id_order_state AND
+                oh.id_order_state = os.id_order_state AND
+                oh.id_order_history = (SELECT MAX(id_order_history) FROM ps_order_history WHERE id_order = o.id_order) AND
+                o.id_order = od.id_order AND
+                od.id_shop = :shop_id AND
+                os.shipped != 1 AND (
+                    o.valid = 1 OR (
+                        os.id_order_state != :error_state AND
+                        os.id_order_state != :cancellation_state
+                    )
+                ) AND sa.id_product = od.product_id AND
+                sa.id_product_attribute = od.product_attribute_id
+                GROUP BY sa.id_product, sa.id_product_attribute
+            )
+        ';
+
+        $updateReservedQuantityQuery = str_replace(
+            array(
+                '{table_prefix}',
+                ':shop_id',
+                ':error_state',
+                ':cancellation_state',
+            ),
+            array(
+                _DB_PREFIX_,
+                (int)$shopId,
+                (int)$errorState,
+                (int)$cancellationState,
+            ),
+            $updateReservedQuantityQuery
+        );
+
+        return Db::getInstance()->execute($updateReservedQuantityQuery);
+    }
+
+    /**
+     * Instance a new StockMvt
+     *
+     * @param null $stockMvtId
+     * @return StockMvt
+     */
+    public function newStockMvt($stockMvtId = null)
+    {
+        if (is_integer($stockMvtId)) {
+            return new StockMvt($stockMvtId);
+        }
+
+        return new StockMvt();
```

Comment:
A complete refacto of this feature should have included the dedicated Doctrine entities.
It's seem you try to reproduce the features of a entity repository (http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/reference/working-with-objects.html#by-primary-key).